pdns/recursordist/rec-main.cc

```diff
@@ -2893,6 +2894,35 @@ int main(int argc, char** argv)
     g_slogout = g_slog->withName("out");
 
     ::arg().setSLog(startupLog);
+
+    if (::arg().mustDo("config")) {
+      string config = ::arg()["config"];
+      if (config == "default") {
+        cout << ::arg().configstring(false, true);
+      }
+      else if (config == "check") {
+        try {
+          if (!::arg().file(configname.c_str())) {
+            SLOG(g_log << Logger::Warning << "Unable to open configuration file '" << configname << "'" << endl,
+                 startupLog->error("No such file", "Unable to open configuration file", "config_file", Logging::Loggable(configname)));
+            exit(1);
+          }
+
+          ::arg().parse(argc, argv);
+          exit(0);
+        }
+        catch (const ArgException& argException) {
+          SLOG(g_log << Logger::Warning << "Unable to parse configuration file '" << configname << "'" << endl,
+               startupLog->error("Cannot parse configuration", "Unable to parse configuration file", "config_file", Logging::Loggable(configname)));
+          exit(1);
+        }
+      }
+      else {
+        cout << ::arg().configstring(true, true);
```

Comment:
Is there an actual difference between `--config` and `--config=default` at this point?

Reply:
Right... Probably should add a call to `::arg().laxFile(...)` before the block to have `--config` dump a loaded configuration file (with warnings about unknown settings). I'm not sure if it's the right way to do it though...

Reply:
Did you take a look how auth does it?